package.json

```
@@ -6,13 +6,15 @@
   "type": "module",
   "scripts": {
     "mvnbuild": "npm run build",
-    "mvntest": "npm test",
+    "mvntest": "eslint --format checkstyle --output-file target/eslint-warnings.xml; vitest run --reporter junit",
```

something like this would mean prettier is still checked which is important
```suggestion
    "mvntest": "eslint --format checkstyle --output-file target/eslint-warnings.xml; vitest run --reporter junit; prettier --check .",
```

---

PS: It works (fails correctly) when running `mvn package` locally:

```
[INFO] --- frontend:1.15.1:npm (npm mvntest) @ pipeline-graph-view ---
[INFO] Running 'npm run mvntest' in /home/felipecrs/repos/pipeline-graph-view-plugin
[INFO] 
[INFO] > pipeline-graph-view-plugin@1.0.0 mvntest
[INFO] > eslint --format checkstyle --output-file target/eslint-warnings.xml; vitest run --reporter junit; npm run format:check
[INFO] 
[INFO] JUNIT report written to /home/felipecrs/repos/pipeline-graph-view-plugin/target/surefire-reports/vitest-junit.xml
[INFO] wslpath: /home/felipecrs/repos/pipeline-graph-view-plugin/node_modules/npm/bin/npm-cli.js
[INFO] 
[INFO] > pipeline-graph-view-plugin@1.0.0 format:check
[INFO] > prettier --check .
[INFO] 
[INFO] Checking formatting...
[INFO] [warn] vitest.config.ts
[INFO] [warn] Code style issues found in the above file. Run Prettier with --write to fix.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  04:32 min
[INFO] Finished at: 2025-04-28T13:45:10-03:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.15.1:npm (npm mvntest) on project pipeline-graph-view: Failed to run task: 'npm run mvntest' failed. org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

Maybe Jenkins is not running `mvn package`?

---

It’s intentionally doing it and then looking at test result files to see the build status

it’s a bit odd but one benefit is it makes it easier to see what failed

---

Is there anything we can do to "fix" that without resorting to #683?

---

I have an idea, one sec.

---

Ok, how about this:

https://github.com/jenkinsci/pipeline-graph-view-plugin/pull/681/commits/349989f0cf0967b9271a7f89451834f2119415a2

It will only "ignore" ESLint and Vitest exit code in Jenkins, but will not be ignored when ran locally.

---

This works great. I hope you like it. :)